samples/nrf9160/nrf_cloud_mqtt_multi_service/src/main.c

```diff
@@ -0,0 +1,59 @@
+/* Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <zephyr.h>
+#include <stdio.h>
+#include <logging/log.h>
+#include "application.h"
+#include "connection.h"
+#include "fota_support.h"
+
+LOG_MODULE_REGISTER(main, CONFIG_MQTT_MULTI_SERVICE_LOG_LEVEL);
+
+
+static void msg_thread_fn(void)
+{
+	/* Continuously consume (that is, send to NRF Cloud) queued-up outgoing messages. */
+	while (true) {
+		(void)consume_device_message();
+	}
+}
+
+static void app_thread_fn(void)
+{
+	/* Wait for first connection before starting the application. */
+	(void)await_connection(K_FOREVER);
+
+	/* Afterwards, allow application to run in this thread. */
+	main_application();
+}
+
+static void conn_thread_fn(void)
+{
+	/* Continuously manage our connection. */
+	manage_connection();
+}
+
+
+K_THREAD_DEFINE(app_thread, CONFIG_APPLICATION_THREAD_STACK_SIZE, app_thread_fn,
```

Comment:
Mainly out of a desire for design symmetry. I feel it makes the structure of the sample slightly more obvious to someone reading it. It is especially nice, since it makes the priority of the application thread explicit, rather than requiring a reader to look up the Zephyr main thread priority. And, since it terminates pretty much immediately, it isn't wasting any appreciable resources, perhaps other than the stack allocated to it.

I do admit it feels a touch strange to have a main function that prints and then terminates. But I think that is outweighed by the explanatory benefits of explicitly declaring the application thread.

If you are not convinced by this argument, though, I will happily move the application thread execution into the main()!

Reply:
Fair enough, no need to change it. The only downside is probably the stack allocation, but in the context of this sample it doesn't matter.